src/figuresClasses.ts

```diff
@@ -1,19 +1,91 @@
-export interface Figure {
+/* eslint-disable max-len */
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape
+  color: Color
+  getArea: () => number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+
+  ) {
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
 
-export class Circle {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('The sides of the triangle must be greater than 0');
+    }
 
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('The longest side of the triangle must be less than the sum of the other two');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape: Shape = 'circle';
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    this.color = color;
+    this.radius = radius;
```

Comment:
You are using [ts constructor assignment](https://kendaleiv.com/typescript-constructor-assignment-public-and-private-keywords/) here. No need to do `this.color = color` because you have `public` in constructor arguments
```suggestion
```